lib/plug/router.ex

```diff
@@ -452,7 +452,8 @@ defmodule Plug.Router do
         conn = update_in unquote(conn).params, merge_params
         conn = update_in conn.path_params, merge_params
 
-        Plug.Conn.put_private(conn, :plug_route, fn var!(conn) -> unquote(body) end)
+        conn = Plug.Conn.put_private(conn, :plug_route, fn var!(conn) -> unquote(body) end)
+        Plug.Conn.put_private(conn, :plug_path, unquote(path))
```

Comment:
I believe we can store both on a single private field in Plug.Conn and then we provide a function for reading it out? Since you are not supposed to access a private field set by a library?

Reply:
Well the whole point of this field is to be able to access it in other plugs, like a plug for metrics or rate limiting, for example. That was my reasoning for initially leaving it out of `private`, since `private` to me meant information set and only used by a framework/library. My goal is instead to provide this information to other libraries, much like `request_path` is available to them on the Plug.Conn.

Can you explain more about how you are thinking this field should be stored? I'm happy to make any changes.

Reply:
It should be stored like this:

    conn = Plug.Conn.put_private(conn, :plug_route, {unquote(path), fn var!(conn) -> unquote(body) end})

and then you should have a function such as `Plug.Router.match_route(conn)` that access the private field and returns only the path (the first element of the tuple).

Reply:
Thanks for the explanation! I've made the changes you suggested. Let me know if this is what you had in mind, or if there are any improvements I can make.